classes/04class/exercises/c04-iac02/gpavelar/output.tf

```diff
@@ -0,0 +1,25 @@
+##
+# Selected VPC
+#
+
+output "vpc_cidr_block" {
+  description = "The CIDR block of the VPC"
+  value       = aws_vpc.devopsacademy-iac.cidr_block
+}
+
+data "aws_subnet_ids" "all_subnets" {
+  vpc_id     = aws_vpc.devopsacademy-iac.id
+  depends_on = [aws_subnet.private-subnet, aws_subnet.public-subnet]
+}
+
+data "aws_subnet" "subnets" {
+  count      = 4
+  id         = tolist(data.aws_subnet_ids.all_subnets.ids)[count.index]
+  depends_on = [aws_subnet.private-subnet, aws_subnet.public-subnet]
+}
+
+output "subnet_cidr_blocks" {
+  value = [for s in data.aws_subnet.subnets : s.cidr_block]
+
+  depends_on = [data.aws_subnet.subnets]
+}
```

Comment:
I didn't see it before. nice job using the `for-in` but I don't think that it is a best practise to add `data` resources to query details from resources that you just created. 
It is an unnecessary extra step because all data is already in the local terraform state.

Just use the variable as you did with `vpc_cidr_block`.